lib/compiler/passes/generate-bytecode.js

```diff
@@ -339,6 +347,8 @@ function generateBytecode(ast) {
           }),
           alternatives.length > 1
             ? buildCondition(
+                // If alternative always match, no need to generate code for the next alternatives
+                -(alternatives[0].match | 0),
```

Comment:
what is the negative or of the first element of alternatives, sub index match, bitwise or of zero?

could you please write code that isn't cryptic?

Reply:
> I think, the comment above explaining enough if you know what for match are.

If people are having to ask you what it means, then this is not correct.

Why are you implementing the same mystery logic in a dozen places?  What happens when someone who isn't you has to maintain this?